packages/formik/src/useField.tsx

```diff
@@ -0,0 +1,252 @@
+export interface FieldInputProps<Value> {
+  /** Value of the field */
+  value: Value;
+  /** Name of the field */
+  name: string;
+  /** Multiple select? */
+  multiple?: boolean;
+  /** Is the field checked? */
+  checked?: boolean;
+  /** Change event handler */
+  onChange: (eventOrValue: React.SyntheticEvent | Value | undefined) => void;
+  /** Blur event handler */
+  onBlur: (eventOrValue: React.SyntheticEvent | boolean | undefined) => void;
+}
+
+export function useField<V = unknown>(
+  options: UseFieldOptions<V>
+): [FieldInputProps<V>, FieldStateAndOperations<V>] {
+  const {
+    name,
+    validate,
+    type,
+    as: is, // as: "select"
+    value: valueProp, // value is special for checkboxes
+    multiple,
+    parse = type && /number|range/.test(type) ? numberParseFn : defaultParseFn,
+    format = defaultFormatFn,
+    formatOnBlur = false,
+  } = options;
+  const { register, unregister, initialValues, forceUpdate } = React.useContext(
+    FormikContext
+  );
+  const [value, setValue, valueRef] = useStateAndRef<V>(
+    options.initialValue || getIn(initialValues, options.name) || ''
+  );
+  const [touched, setTouched, touchedRef] = useStateAndRef<boolean>(false);
+  const [error, setError, errorsRef] = useStateAndRef<string | undefined>(
+    undefined
+  );
+  React.useEffect(() => {
+    register(name, {
+      valueRef,
+      touchedRef,
+      errorsRef,
+      setError,
+      setValue,
+      setTouched,
+      validate,
+    });
+    return () => {
+      unregister(name);
+    };
+  }, [
+    name,
+    validate,
+    register,
+    unregister,
+    setValue,
+    setTouched,
+    setError,
+    errorsRef,
+    touchedRef,
+    valueRef,
+  ]);
+
+  const getValueFromEvent = useEventCallback(
+    (event: React.SyntheticEvent<any>) => {
+      // React Native/Expo Web/maybe other render envs
+      if (
+        !isReactNative &&
+        event.nativeEvent &&
+        (event.nativeEvent as any).text !== undefined
+      ) {
+        return (event.nativeEvent as any).text;
+      }
+      // React Native
+      if (isReactNative && event.nativeEvent) {
+        return (event.nativeEvent as any).text;
+      }
+      const target = event.target ? event.target : event.currentTarget;
+      const { type, value: valueProp, checked, options, multiple } = target;
+      return /checkbox/.test(type) // checkboxes
+        ? getValueForCheckbox(value as $FixMe, checked, valueProp)
+        : !!multiple // <select multiple>
+        ? getSelectedValues(options)
+        : valueProp;
+    }
+  );
+  const handleChange = useEventCallback(eventOrValue => {
+    if (isInputEvent(eventOrValue)) {
+      if (eventOrValue.persist) {
+        eventOrValue.persist();
+      }
+      setValue(parse(getValueFromEvent(eventOrValue), name));
+    } else {
+      setValue(parse(eventOrValue, name));
+    }
+    if (validate) {
+      const err = validate(eventOrValue.target.value);
+      setError(err);
+    }
+    forceUpdate();
```

Comment:
Since there is no longer any top-level state, is this call to `forceUpdate` needed on each change?